applications/StructuralMechanicsApplication/custom_elements/total_lagrangian_mixed_volumetric_strain_element.h

```diff
@@ -0,0 +1,644 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:         BSD License
+//                   license: structural_mechanics_application/license.txt
+//
+//  Main authors:    Ruben Zorrilla
+//                   Riccardo Rossi
+//                   Guglielmo Scovazzi
+//
+
+#pragma once
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/define.h"
+#include "includes/element.h"
+#include "utilities/integration_utilities.h"
+
+// Application includes
+#include "structural_mechanics_application_variables.h"
+
+namespace Kratos
+{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/**
+ * @class TotalLagrangianMixedVolumetricStrainElement
+ * @ingroup StructuralMechanicsApplication
+ * @brief Small displacement with strain based mixed formulation element
+ * @details This implements a total Lagrangian element formulation with an extra Jacobian determinant nodal DOF
+ * @author Ruben Zorrilla
+ * @author Riccardo Rossi
+ * @author Guglielmo Scovazzi
+ */
+template<std::size_t TDim>
+class KRATOS_API(STRUCTURAL_MECHANICS_APPLICATION) TotalLagrangianMixedVolumetricStrainElement
+    : public Element
+{
+
+    static constexpr std::size_t NumNodes = TDim + 1;
+    static constexpr std::size_t StrainSize = TDim == 2 ? 3 : 6;
+    static constexpr std::size_t BlockSize = TDim +1;
+    static constexpr std::size_t LocalSize = NumNodes*BlockSize;
+
+protected:
+
+    /**
+     * Internal variables used in the kinematic calculations
+     */
+    struct KinematicVariables
+    {
+        double detF;
+        Matrix F;
+        double detJ0;
+        Matrix J0;
+        Matrix InvJ0;
+        Vector N;
+        Matrix DN_DX;
+        BoundedMatrix<double,NumNodes, TDim> Displacements;
+        BoundedVector<double,NumNodes> JacobianDeterminant;
+        Vector EquivalentStrain;
+
+        KinematicVariables()
+        {
+            detF = 1.0;
+            detJ0 = 1.0;
+            F = IdentityMatrix(TDim);
+            J0 = ZeroMatrix(TDim, TDim);
+            InvJ0 = ZeroMatrix(TDim, TDim);
+            N = ZeroVector(NumNodes);
+            DN_DX = ZeroMatrix(NumNodes, TDim);
+            Displacements = ZeroMatrix(NumNodes, TDim);
+            JacobianDeterminant = ZeroVector(NumNodes);
```

Comment:
just for my curiosity, is it faster to do this or a `resize()` and then `clear()`?

Reply:
I think it is pretty much the same. The point in here is that we're not using bounded arrays (as pointed by @loumalouomega) so we need to resize one way or another.